lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -401,12 +400,40 @@ collectOpsToPipeline(scf::ForOp forOp,
         loadInfo.sharedEncoding =
             getSharedEncIfAllUsersAreDotEnc(loadOp.getResult())
                 .value_or(nullptr);
+
+        // HACK: Triton LLVM codegen has a bug where local_loads from #shared to
+        // #mma layout can lead to invalid code if the loaded shape is smaller
+        // than the mma tile (e.g. loading a 128x1 tensor for an MMAv2 dot with
+        // tile {16,8} is bad because 1 < 8).  To work around this, don't
+        // pipeline such loads.
+        //
+        // The codegen bug is caught by an assertion, so if you think you've
+        // fixed it, feel free to delete this code and see if the assert still
+        // fails.  :)
+        if (!loadInfo.sharedEncoding) {
+          if (auto dotEnc = dot.getResult()
+                                .getType()
+                                .getEncoding()
+                                .dyn_cast<ttg::NvidiaMmaEncodingAttr>()) {
+            auto loadTy = loadOp.getType().cast<RankedTensorType>();
+            auto mmaInstrShape = dotEnc.getInstrShape();
+            if (loadTy.getRank() < mmaInstrShape.size())
+              continue;
+            bool ok = true;
+            for (int i = 0; i < mmaInstrShape.size(); i++) {
+              if (loadTy.getShape()[loadTy.getRank() - mmaInstrShape.size() +
+                                    i] != mmaInstrShape[i]) {
```

Comment:
so the shape loaded as to match exactly the mma instr shape? That seems like an odd condition. Should the condition be that it need to be a multiple of it?

Reply:
Yikes, sorry.  That was an autocomplete bug.  Should be less-than, which is equivalent to divisible-by because everything is a power of 2.